support/builder/s3migrate.sh

```diff
@@ -0,0 +1,311 @@
+#!/bin/bash
+set -euo pipefail
```

Comment:
I assume we've run shellcheck or other linters on the script, if not, would be good to run those..

Reply:
Yep, shellcheck full-up has been fully run and corrected.

Reply:
👍